oauth2client/flask.py

```diff
@@ -0,0 +1,299 @@
+    """Raised when OAuth2 settings are not adequately configured."""
+    pass
+
+
+class NoCredentialsError(Exception):
+    """Raised when attempting to use credentials that do not exist."""
+    pass
+
+
+class OAuth2(object):
+    """Flask extension for making OAuth 2.0 easier.
+
+    The extension includes views that handle the entire auth flow and a
+    @required decorator to automatically ensure that user credentials are
+    available.
+
+    To configure::
+
+        app = Flask(__name__)
+
+        app.config['SECRET_KEY'] = 'your-secret-key'
+        app.config['OAUTH2_CLIENT_SECRETS_JSON'] = 'client_secrets.json'
+
+        oauth2 = OAuth2(app)
+
+
+    To use::
+
+        @app.route('/needs_credentials')
+        @oauth2.required
+        def example():
+            # http is authorized with the user's credentials and can be used
+            # to make http calls.
+            http = oauth2.http()
+
+            # Or, you can access the credentials directly
+            credentials = oauth2.credentials
+
+
+        @app.route('/info')
+        @oauth2.required
+        def info():
+            return "Hello, {}".format(oauth2.email)
+
+    """
+
+    def __init__(self, app=None, *args, **kwargs):
+        self.app = app
+        if app is not None:
+            self.init_app(app, *args, **kwargs)
+
+    def init_app(self, app, scopes=None, client_secrets_file=None,
+                 client_id=None, client_secret=None, authorize_callback=None,
+                 logout_callback=None):
+        self.app = app
+
+        if not scopes:
+            scopes = app.config.get('OAUTH2_SCOPES', DEFAULT_SCOPES)
+
+        self.scopes = scopes
+
+        self.client_id, self.client_secret = client_id, client_secret
+
+        if not self.client_id or not self.client_secret:
+            if client_secrets_file:
+                self.client_id, self.client_secret = self._load_client_secrets(
+                    client_secrets_file)
+            elif 'OAUTH2_CLIENT_SECRETS_JSON' in app.config:
+                self.client_id, self.client_secret = self._load_client_secrets(
+                    app.config['OAUTH2_CLIENT_SECRETS_JSON'])
+            else:
+                self.client_id, self.client_secret = self._load_app_config(app)
+
+        app.register_blueprint(self._create_blueprint())
+
+        self.authorize_callback = authorize_callback
+        self.logout_callback = logout_callback
+
+        app.after_request(self._update_session)
+
+    def _load_client_secrets(self, filename):
+        client_type, client_info = clientsecrets.loadfile(filename)
+        if client_type != clientsecrets.TYPE_WEB:
+            raise ConfigurationError(
+                "The flow specified in %s isn't supported." % client_type)
+
+        return client_info['client_id'], client_info['client_secret']
+
+    def _load_app_config(self, app):
+        if ('OAUTH2_CLIENT_SECRET' not in app.config or
+                'OAUTH2_CLIENT_ID' not in app.config):
```

Comment:
1. something, something, line isn't distinguishable from next line.
2. In flask extensions the configuration names generally aren't constants.